src/v/config/property.h

```diff
@@ -113,9 +113,13 @@ class property : public base_property {
     // serialize the value. the key is taken from the property name at the
     // serialization point in config_store::to_json to avoid users from being
     // forced to consume the property as a json object.
-    void to_json(json::Writer<json::StringBuffer>& w, redact_secrets)
+    void to_json(json::Writer<json::StringBuffer>& w, redact_secrets redact)
       const override {
-        json::rjson_serialize(w, _value);
+        if (is_secret() && redact == redact_secrets::yes) {
+            json::rjson_serialize(w, secret_placeholder);
```

Comment:
Can we tweak this to only redact when it's non-empty?

It's a nicer experience if the `rpk cluster config edit` on a fresh system shows blank for the cloud key, that way the user can be confident when setting it that they're not overwriting anything.

I think it's correct to say that the value of the key may be secret, but the fact of whether it is set is not.

Reply:
Sure. Do you think this should be limited to emptiness, or all defaults?

Reply:
I think just emptiness is fine, I don't anticipate anything being secret but also having a default

Reply:
You're right; as it turns out, all our current secret defaults are empty. I updated to redact based on `is_default()` as is done for logging.